### PR TITLE
Set registry to managed state after OCP deployment

### PR DESCRIPTION
**Pull request by petr-balogh**
In order to fix stage deployment on OCP 4.3 we need to move change of
registry to managed state after the deployment of OCP.

Fixes: #1706

Signed-off-by: Petr Balogh <pbalogh@redhat.com>

### CODE REVIEW

**Comment by petr-balogh** [comment]
Running verification job here:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/5917/console

**Comment by petr-balogh** [comment]
Previous job failed cause of 4.3 is not pushed to the stage jet.

I've tried to do some manual cleanup and running on the same cluster here with OCS 4.2:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/5925/console

File: ocs_ci/deployment/deployment.py [hunk 1]
```diff
@@ -164,6 +164,27 @@ def post_ocp_deploy(self):
...
+        # properly propagated for the stage deployment as mentioned in BZ.
+        if(config.ENV_DATA['platform'] not in constants.CLOUD_PLATFORMS):
+            run_cmd(
+                f'oc patch {constants.IMAGE_REGISTRY_CONFIG} --type merge -p '
+                f'\'{{"spec":{{"storage": {{"emptyDir":{{}}}}}}}}\''
+            )
+            run_cmd(
+                f'oc patch {constants.IMAGE_REGISTRY_CONFIG} --type merge -p '
+                f'\'{{"spec":{{"managementState": "Managed"}}}}\''
+            )
```
Comment (clacroix12) [inline ocs_ci/deployment/deployment.py]: This is quite a lot of information for individual comment lines. Would it be better to create a function that we call here instead and have this information in the docstring? This has the additional benefit of propagating this information to our documentation.

Comment (petr-balogh) [reply ocs_ci/deployment/deployment.py]: Thanks for suggestion. I incorporated this in next commit.

**Comment by petr-balogh** [comment]
Verification job scheduled here on one of vmware cluster:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/5943/console

I will check its status tomorrow morning.

**Comment by petr-balogh** [comment]
Deployment passed - Verified

**Pull request merged**